ouroboros-consensus-cardano/tools/db-analyser/Main.hs

```diff
@@ -1,307 +1,109 @@
+  }
+
+data BlockType =
+    ByronBlock   ByronBlockArgs
+  | ShelleyBlock ShelleyBlockArgs
+  | CardanoBlock CardanoBlockArgs
 
 {-------------------------------------------------------------------------------
-  Command line args
+  Parsing
 -------------------------------------------------------------------------------}
 
-data CmdLine = CmdLine {
-      clConfig    :: [FilePath]
-    , clIsMainNet :: Bool
-    , clImmDB     :: FilePath
-    , clBlockType :: Maybe BlockType
-    , clAnalysis  :: AnalysisName
-    } deriving Show
-
-data BlockType = Byron | Shelley | Cardano
-  deriving Show
-
 parseCmdLine :: Parser CmdLine
 parseCmdLine = CmdLine
-    <$> many (strOption (mconcat [
-            long "config"
-          , help "Path to config file or files. Multiple occurences of the\
-                 \option can be used"
-          , metavar "PATH"
-          ]))
-    <*> flag True False (mconcat [
-            long "testnet"
-          , help "The DB contains blocks from testnet rather than mainnet"
-          ])
-    <*> strOption (mconcat [
+    <$> strOption (mconcat [
             long "db"
-          , help "Path to the chain DB (parent of \"immutable\" directory)"
+          , help "Path to the chain DB"
           , metavar "PATH"
           ])
-    <*> parseBlockType
+    <*> flag False True (mconcat [
+            long "verbose"
+          , help "Enable verbose logging"
+          ])
+    <*> flag False True (mconcat [
+            long "onlyImmDB"
+          , help "Validate only the immutable DB (e.g. do not do ledger validation)"
+          ])
+    <*> parseImmValidationPolicy
+    <*> parseVolValidationPolicy
+    <*> blockTypeParser
     <*> parseAnalysis
 
-parseBlockType :: Parser (Maybe BlockType)
-parseBlockType = asum [
-      flag' (Just Byron) $ mconcat [
-          long "byron"
-        , help "A Byron network"
+parseImmValidationPolicy :: Parser (Maybe ImmDB.ValidationPolicy)
+parseImmValidationPolicy = parseMaybe $ asum [
+      flag' ImmDB.ValidateMostRecentChunk $ mconcat [
+          long "imm-validate-most-recent"
+        , help "Validate only the most recent chunk of the Immutable db"
         ]
-    , flag' (Just Shelley) $ mconcat [
-          long "shelley"
-        , help "A Shelley network"
+    , flag' ImmDB.ValidateAllChunks $ mconcat [
+          long "imm-validate-all-chunks"
+        , help "Validate all chunks of the Immutable db"
         ]
-    , flag' (Just Cardano) $ mconcat [
-          long "cardano"
-        , help "A Byron-to-Shelley network"
+    ]
+
+parseVolValidationPolicy :: Parser (Maybe VolDB.BlockValidationPolicy)
+parseVolValidationPolicy = parseMaybe $ asum [
+      flag' VolDB.NoValidation $ mconcat [
+          long "vol-no-validation"
+        , help "Don't validate blocks of the Volatile db"
+        ]
+    , flag' VolDB.ValidateAll $ mconcat [
+          long "vol-validate-all"
+        , help "Validate all blocks of the Volatile db"
         ]
-    , pure Nothing
     ]
```

Comment:
Let's simplify this into `parseFullValidation :: Parser Bool`, which defaults to `False`, but when `--full-validation` is specified, the `ImmDB.ValidateMostRecentChunk` and `VolDB.ValidateAll` policies are used (similar to [what we do for the real node](https://github.com/input-output-hk/ouroboros-network/blob/aecfe77de784e6c75108172802ee59d2d7087eaf/ouroboros-consensus/src/Ouroboros/Consensus/Node.hs#L184)). You can use "Validate all blocks on disk" as the `help` text.

Hmm, I'd still like to default it to `True` in case of `OnlyValidation` without losing the ability to disable it explicitly. Maybe we need `Maybe True` for that, with `Nothing` as default and `--no-full-validation` for `Just False`.